src/client/datascience/jupyter/kernels/cellExecution.ts

```diff
@@ -185,7 +185,8 @@ export class CellExecution {
             `Cell Exec contents ${this.cell.document.getText().substring(0, 50)}...`
         );
         if (!this.canExecuteCell()) {
-            this.execution?.end({});
+            // End state is bool | undefined not optional. Undefined == not success or failure
```

Comment:
The new tristate here for success is a bit funky so I called it out with a comment.